source/unified-test-format/unified-test-format.md

```diff
@@ -1416,6 +1428,53 @@ the resulting change stream might be saved with [operation.saveResultAsEntity](#
 Test runners MUST NOT iterate the change stream when executing this operation and test files SHOULD NOT specify
 [operation.expectResult](#operation_expectResult) for this operation.
 
+#### clientBulkWrite
+
+These considerations only apply to the `MongoClient.bulkWrite` method. See [bulkWrite](#bulkwrite) for special considerations for `MongoCollection.bulkWrite`.
+
+The `models` parameter for `clientBulkWrite` is documented as a list of WriteModel interfaces. Each WriteModel
+implementation (e.g. InsertOneModel) provides important context to the method, but that type information is not easily
+expressed in YAML and JSON. To account for this, test files MUST nest each WriteModel object in a single-key object,
+where the key identifies the request type (e.g. "insertOne") and its value is an object expressing the parameters, as in
+the following example:
+
+```
+arguments:
+  models:
+    - insertOne:
+        document: { _id: 1, x: 1 }
+    - replaceOne:
+        filter: { _id: 2 }
+        replacement: { x: 2 }
+        upsert: true
+    - updateOne:
+        filter: { _id: 3 }
+        update: { $set: { x: 3 } }
+        upsert: true
+    - updateMany:
+        filter: { }
+        update: { $inc: { x: 1 } }
+    - deleteOne:
+        filter: { x: 2 }
+    - deleteMany:
+        filter: { x: { $gt: 2 } }
+  ordered: true
+```
+
+Because the `insertResults`, `updateResults`, and `deleteResults` may be absent or empty in the `BulkWriteResult` returned
+from a summary-only bulk write, the `clientBulkWrite` operation MUST use the [$$unsetOrMatches](#unsetormatches) operator
```

Comment:
Note: this spec typically uses SHOULD instead of MUST, particular for things that cannot be enforced via the JSON schema. That's explained in the Design Rationale section.

This would be a candidate for SHOULD.

Reply:
makes sense, updated